`scoot.arithmetic`: Interpreter for arithmetic expressions and programs

octalsrc opened the pull request:
We have completed the interpreter for arithmetic expressions and programs consisting of sequences of `AssignStmt` that use arithmetic expressions.

The expression interpreter is probably exactly what we want, but the statement-sequence interpreter will need to become more general and will likely be moved up into the `scoot` package.  But that's the next task.

ftc committed:
soot loading

ftc committed:
jimple body exists now

ftc committed:
test files

ftc committed:
moved test resources to keep travis happy

ftc committed:
clean up comments

Commit:
fix soot class path setup (now it works not only for *nix, but also for Windows)

Commit:
initial commit for jimple interpreter

Commit:
work in progress

Commit:
well... the concrete interpreter seems to be working now

Commit:
add test case

Commit:
try to fix test failure (because of loading soot twice in the test suite)

Commit:
comment out shawn's test case, since soot cannot be loaded twice and the success of my test covers the success of his

octalsrc committed:
starting with tests

Not sure how to proceed:

What is our equivalent of interpret("1 + 1"), for example?

Commit:
deprecate methods in jimple interpreter

ftc committed:
Merge pull request #16 from cuplv/jimple-interpreter

Jimple interpreter

octalsrc committed:
strange error - soot Exprs won't nest?

octalsrc committed:
Merge remote-tracking branch 'origin/jimple-interpreter' into jimple-arithmetic-tests

octalsrc committed:
switched to public Jimple methods in Builder

octalsrc committed:
new builder for test cases

octalsrc committed:
Merge branch 'develop' into scoot-arithmetic

Commit:
work in progress

Commit:
could not figure out how to transform from type (Int, Int) => Boolean to type (Int, Int) => Int

Commit:
could not figure out how to transform from type (Int, Int) => Boolean to type (Int, Int) => Int

ftc committed:
concurrent rerun fix

ftc committed:
merge tianhan branch and conc fix re enable test

octalsrc committed:
Fix bool/int case in arithmetic interpreter

Really our interpreter should interpret expressions of different types
than just Int (so it could return an Int *or* a Bool), so a comparison
on Ints should return a Bool.  But we can figure out what to do about
that later; right now we are only handling core integer arithmetic.

octalsrc committed:
Remove deprecated `jimpleinterpreter` and associated tests

This code was causing errors due to duplication of `sootloading` code
in its test suite.  Its functionality is being replaced by the `scoot`
package.

octalsrc committed:
Merge remote-tracking branch 'origin/soot-setup' into soot-loading-test

octalsrc committed:
Implement most of int arithmetic interpreter

octalsrc committed:
Redefine Scoot Value tests

octalsrc committed:
Complete Scoot arithmetic interpreter, tests included :)

octalsrc committed:
Merge remote-tracking branch 'origin/develop' into develop-merge-test

octalsrc committed:
Fix doc typo

octalsrc committed:
Merge remote-tracking branch 'origin/develop' into merge-develop-test

jpw234 commented:
The Soot loading stuff still looks kinda wonky but I don't think that's the point of this PR anyway, right? The interpreter looks good.

octalsrc committed:
Merge remote-tracking branch 'origin/develop' into merge-test

octalsrc committed:
Remove soot-loading things (not relevant to this PR)

This branch only uses the jars, and we don't have consensus on adding
the soot-loading stuff yet, so I'm removing it from this branch for
now.

octalsrc committed:
Address codacy complaint (switch to wildcard import)

octalsrc commented:
I've removed the soot-loading parts from this pull-request, since it was only using the jars.
